aws/provision.sh

```diff
@@ -14,6 +17,53 @@ chmod 400 ~/.ssh/id_rsa.pub
 eval `ssh-agent`
 ssh-add ~/.ssh/id_rsa
 
-wget https://s3.amazonaws.com/bosh-init-artifacts/bosh-init-0.0.72-linux-amd64 -O bosh-init
-chmod +x bosh-init
-./bosh-init deploy manifest_aws.yml
+# TODO: download bosh-init from our own bucket
+if [ ! -x bosh-init ]; then
+	wget https://s3.amazonaws.com/bosh-init-artifacts/bosh-init-0.0.72-linux-amd64 -O bosh-init
+	chmod +x bosh-init
+fi
+export BOSH_INIT_LOG_LEVEL=debug
+export BOSH_INIT_LOG_PATH=bosh_init.log
+time ./bosh-init deploy manifest_aws.yml
+
+# TODO: install bosh cli rubygems in parallel to bosh-init (which mainly stresses bosh server)
+if gem list | grep -q bosh_cli; then
+  echo "Bosh cli already installed, skipping..."
+else
+  echo "Installing bosh cli..."
+  time sudo gem install bosh_cli -v 1.3056.0 --no-ri --no-rdoc
+fi
+
+export PATH=$PATH:/usr/local/bin/bosh
+
+# FIXME: make passwords properly strong
+# FIXME: bosh for some reason ignores name and sets it to "my-bosh"
+# FIXME: cmd below works, but bosh compains 'stty: standard input: Inappropriate ioctl for device'
+echo -e "admin\nadmin" | bosh target 10.0.0.6 "IamIgnored"
+
+# TODO: download stemcell from our own bucket using multiple threads, in paralell with other tasks
+if [ ! -f $STEMCELL ]; then
+  time bosh download public stemcell $STEMCELL
+fi
+time bosh upload stemcell $STEMCELL --skip-if-exists
+
+# TODO: this can also happen in parallel
+if [ ! -d cf-release ]; then
+  time git clone https://github.com/cloudfoundry/cf-release.git
+fi
+
+# TODO: pre-seed bosh cache, otherwise action below takes forever...
+# FIXME: make output from bosh command stream and have color. It takes quite some time to complete, immediate output is desirable.
+# (this script is run via ssh from makefile)
+
+# No, you can't bosh upload release cf-release/releases/cf-215.yml, you have to CD into the cf-release first
+echo "Uploading v215 release to bosh..."
+cd cf-release
+git checkout v215
+./update
+time bosh upload release releases/cf-215.yml
+# Run deployment
+cd ~
+sed -i "s/BOSH_UUID/$(bosh status --uuid)/" cf-manifest.yml
+bosh deployment cf-manifest.yml
+time bosh deploy
```

Comment:
If you do a 'bosh -n deploy' here instead of just 'bosh deploy' then it will be in non-interactive mode and will not prompt the user to answer 'yes' to deploying.